app/src/main/java/org/schabi/newpipe/fragments/MainFragment.java

```diff
@@ -189,6 +196,46 @@ private void updateTabsIconAndDescription() {
     private void updateTitleForTab(final int tabPosition) {
         setTitle(tabsList.get(tabPosition).getTabName(requireContext()));
     }
+    private void updateTabsPosition() {
+        final ScrollableTabLayout tabLayout = binding.mainTabLayout;
+        final ViewPager viewPager = binding.pager;
+        final RelativeLayout.LayoutParams tabParams = (RelativeLayout.LayoutParams)
+                tabLayout.getLayoutParams();
+        final RelativeLayout.LayoutParams pagerParams = (RelativeLayout.LayoutParams)
+                viewPager.getLayoutParams();
+        if (PreferenceManager.getDefaultSharedPreferences(requireContext())
+                .getBoolean(getString(R.string.main_tabs_position_key), false)) {
+            tabParams.removeRule(RelativeLayout.ALIGN_PARENT_TOP);
+            tabParams.addRule(RelativeLayout.ALIGN_PARENT_BOTTOM);
+            pagerParams.removeRule(RelativeLayout.BELOW);
+            pagerParams.addRule(RelativeLayout.ABOVE, R.id.main_tab_layout);
+            tabLayout.setSelectedTabIndicatorGravity(TabLayout.INDICATOR_GRAVITY_TOP);
+            final TypedValue typedValueBackground = new TypedValue();
+            getContext().getTheme().resolveAttribute(R.attr.colorSecondary, typedValueBackground,
+                    true);
+            tabLayout.setBackgroundColor(typedValueBackground.data);
+            tabLayout.setTabRippleColor(ColorStateList.valueOf(
+                    getResources().getColor(R.color.gray)));
+            tabLayout.setTabIconTint(ColorStateList.valueOf(getResources().getColor(R.color.gray)));
+            tabLayout.setSelectedTabIndicatorColor(getResources().getColor(R.color.gray));
+        } else {
+            tabParams.removeRule(RelativeLayout.ALIGN_PARENT_BOTTOM);
+            tabParams.addRule(RelativeLayout.ALIGN_PARENT_TOP);
+            pagerParams.removeRule(RelativeLayout.ABOVE);
+            pagerParams.addRule(RelativeLayout.BELOW, R.id.main_tab_layout);
+            tabLayout.setSelectedTabIndicatorGravity(TabLayout.INDICATOR_GRAVITY_BOTTOM);
+            final TypedValue typedValue = new TypedValue();
+            getContext().getTheme().resolveAttribute(R.attr.colorPrimary, typedValue,
+                    true);
+            tabLayout.setBackgroundColor(typedValue.data);
+            tabLayout.setTabRippleColor(binding.mainTabLayout.getTabRippleColor().withAlpha(32));
+            tabLayout.setTabIconTint(binding.mainTabLayout.getTabIconTint());
+            tabLayout.setSelectedTabIndicatorColor(ContextCompat
+                    .getColor(requireContext(), R.color.white));
```

Comment:
Please note that `binding.mainTabLayout` is the exact same thing as `tabLayout`. Therefore the first two lines either do not do anything, or creates the following situation after switching the setting two times.
![image](https://user-images.githubusercontent.com/36421898/217217574-72e02425-1ea8-403e-bbc8-d6350b2aece9.png)

Here it would be ok to use white everywhere as that seems like what's being done in `fragment_main.xml`.